Doc/library/devmode.rst

```diff
@@ -0,0 +1,184 @@
+.. _devmode:
+
+Python Development Mode
+=======================
+
+.. versionadded:: 3.7
+
+The Python Development Mode introduces additional runtime checks which are too
+expensive to be enabled by default. It should not be more verbose than the
+default if the code is correct: new warnings are only emitted when an issue is
+detected.
+
+It can be enabled using :option:`-X dev <-X>` command line option or by setting
+:envvar:`PYTHONDEVMODE` environment variable to ``1``.
+
+Effects of the developer mode
+=============================
+
+Enabling the development mode has multiple effects:
+
+* Add ``default`` :ref:`warning filter <describing-warning-filters>`, as
+  :option:`-W default <-W>` command line option. The following warnings are
+  shown, whereas they are ignored by the default warning filters:
+
+  * :exc:`DeprecationWarning`;
+  * :exc:`ImportWarning`;
+  * :exc:`PendingDeprecationWarning`;
+  * :exc:`ResourceWarning`.
+
+  Use :option:`-W error <-W>` command line option or set
+  :envvar:`PYTHONWARNINGS` environment variable to ``error`` to treat warnings
+  as errors.
+
+* Install debug hooks on memory allocators to check for:
+
+  * buffer underflow;
+  * buffer overflow
+  * memory allocator API violation;
+  * usage of the GIL.
+
+  See the :c:func:`PyMem_SetupDebugHooks` C function. Similar to setting
+  :envvar:`PYTHONMALLOC` environment variable to ``debug``.
+
+  To enable the development without installing debug hooks on memory
+  allocators, set :envvar:`PYTHONMALLOC` environment variable to ``default``.
+
+* Call :func:`faulthandler.enable` at Python startup to enable the
+  :mod:`faulthandler` module: install handlers for the :const:`SIGSEGV`,
```

Comment:
Does calling `faulthandler.enable` install the handlers?  If so, it might be better to say `which installs` instead of `: install`.